src/bespokelabs/curator/prompter/prompter.py

```diff
@@ -286,25 +288,157 @@ def _completions(
         return dataset
 
 
-def _get_function_hash(func) -> str:
-    """Get a hash of a function's source code."""
-    if func is None:
-        return xxh64("").hexdigest()
+class PathIndependentPickler:
+    """A custom serializer that ensures consistent function serialization."""
 
-    file = BytesIO()
-    dill.Pickler(file, recurse=True).dump(func)
-    return xxh64(file.getvalue()).hexdigest()
+    def __init__(self, file):
+        """Initialize the serializer."""
+        self.file = file
+        self._debug = []  # Store debug information
 
+    def _serialize_code(self, code):
+        """Serialize only the essential parts of a code object."""
+        return {
+            "bytecode": code.co_code.hex(),  # bytecode as hex string
+            "constants": [
+                (
+                    self._serialize_code(c)
+                    if isinstance(c, types.CodeType)
+                    else (list(c) if isinstance(c, (tuple, set, frozenset)) else c)
+                )
+                for c in code.co_consts
+            ],
+            "names": sorted(code.co_names),
+            "varnames": sorted(code.co_varnames),
+            "freevars": sorted(code.co_freevars),
+            "cellvars": sorted(code.co_cellvars),
+            "argcount": code.co_argcount,
+            "posonlyargcount": code.co_posonlyargcount,
+            "kwonlyargcount": code.co_kwonlyargcount,
+            "nlocals": code.co_nlocals,
+            "stacksize": code.co_stacksize,
+            "flags": code.co_flags & ~0b111111111111111,  # Clear position-dependent flags
```

Comment:
you don't need to clear flags here. i think you can take them as-is since we do want to have some differentiation between nested vs non-nested.